flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroupTest.java

```diff
@@ -44,4 +52,132 @@ protected QueryScopeInfo createQueryServiceMetricInfo(CharacterFilter filter) {
+		config.setString(ConfigConstants.METRICS_REPORTER_PREFIX + "test2." + ConfigConstants.METRICS_REPORTER_SCOPE_DELIMITER, "!");
+
+
+		MetricRegistry testRegistry = new MetricRegistryTest(MetricRegistryConfiguration.fromConfiguration(config));
+		TaskManagerMetricGroup tmGroup = new TaskManagerMetricGroup(testRegistry, "host", "id");
+		tmGroup.counter(1);
+		testRegistry.shutdown();
+		assert TestReporter1.countSuccessChecks == 4;
+	}
+
+	@Test
+	public void filteringForNullReporters() {
+		MetricRegistryTest.countSuccessChecks = 0;
+		Configuration config = new Configuration();
+		config.setString(ConfigConstants.METRICS_SCOPE_NAMING_TM, "A.B.C.D");
+		MetricRegistry testRegistry = new MetricRegistryTest(MetricRegistryConfiguration.fromConfiguration(config));
+		TaskManagerMetricGroup tmGroupForTestRegistry = new TaskManagerMetricGroup(testRegistry, "host", "id");
+		assert testRegistry.getReporters().size() == 0;
+		tmGroupForTestRegistry.counter(1);
+		testRegistry.shutdown();
+		assert MetricRegistryTest.countSuccessChecks == 1;
+	}
+
+	public static class TestReporter1 extends TestReporter {
+		protected static int countSuccessChecks = 0;
+		@Override
+		public String filterCharacters(String input) {
+			return input.replace("A", "RR");
+		}
+		@Override
+		public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
+			assertEquals("A-B-C-D-1", group.getMetricIdentifier(metricName));
+			// ignore all next filters for scope -  because scopeString cached with only first filter
+			assertEquals("A-B-C-D-1", group.getMetricIdentifier(metricName, staticCharacterFilter));
+			assertEquals("A-B-C-D-1", group.getMetricIdentifier(metricName, this));
+			assertEquals("A-B-C-D-4", group.getMetricIdentifier(metricName, new CharacterFilter() {
+				@Override
+				public String filterCharacters(String input) {
+					return input.replace("B", "RR").replace("1", "4");
+				}
+			}));
+			countSuccessChecks++;
+		}
+	}
+	public static class TestReporter2 extends TestReporter1 {
+		@Override
+		public String filterCharacters(String input) {
+			return input.replace("B", "RR");
+		}
+		@Override
+		public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
+			assertEquals("A!RR!C!D!1", group.getMetricIdentifier(metricName, this));
+			// ignore all next filters -  because scopeString cached with only first filter
+			assertEquals("A!RR!C!D!1", group.getMetricIdentifier(metricName));
+			assertEquals("A!RR!C!D!1", group.getMetricIdentifier(metricName, staticCharacterFilter));
+			assertEquals("A!RR!C!D!3", group.getMetricIdentifier(metricName, new CharacterFilter() {
+				@Override
+				public String filterCharacters(String input) {
+					return input.replace("A", "RR").replace("1", "3");
+				}
+			}));
+			countSuccessChecks++;
+		}
+	}
+	static class MetricRegistryTest extends MetricRegistry {
+		protected static int countSuccessChecks = 0;
```

Comment:
Any error in ``checkMethod``  is calling fail test. We can't get some errors from the ``super.register`` because all errors is logging in it. 
That's why I think that this field is not necessary.